Sources/LeafKit/LeafParser.swift

```diff
@@ -223,9 +228,16 @@ struct LeafParser {
         // .parametersStart - start parameters
         // .tagIndicator - a new tag started
         switch next {
-        case .raw:
+        // MARK: no param, no body case should be re-evaluated?
+        // we require that tags have parameter notation INSIDE parameters even when they're
+        // empty - eg `#tag(anotherTag())` - so `#anotherTag()` should be required, not
+        // `#anotherTag`. If that's enforced, the only acceptable non-decaying noparam/nobody
+        // use would be `#endTag` to close a body
+        case .raw,
```

Comment:
### PLEASE READ ###
I would REALLY like to iron out this inconsistency. If a tag can exist with no parameters or body; eg, a `date` tag that returns current date if nothing is specified, it can be written as a top-level tag as `#date`. Inside parameters, we do NOT allow `(date)` - only `(date())`. THIS IS CONFUSING! and in my opinion, should not be allowed. The *ONLY* situation in which a top-level tag should be usable without parens is for end tags - eg `#endfor` where it is explicitly ending a tag body.

Reply:
I agree. To use a top level tag it should be `#date()`. I _think_ this is how it worked in Leaf 3, too, but I'm not sure. `#end` is a special case and does not need `()`.

Reply:
Cool - I wasn't sure since branching like this *was* in the codebase - but would be very happy to isolate it as the only acceptable pattern. Didn't see any examples of no-parameter tags being used but there's certainly use for it, just should be specified as an explicit-no-parameter-with-parens requirement.

Reply:
Will adjust this in next PR